main_loop_tf/main.py

```diff
@@ -730,19 +843,39 @@ def main_loop(placeholders, val_placeholders, train_outs, train_summary_op,
             # TODO: It should be IoU or any other metric, but in this
             # case our loss is Dice Coefficient so it's fine
             loss_value = -1.0 if loss_value < -cfg.thresh_loss else loss_value
-            in_values = [x_in, y_in, split_dim, labels_split_dim,
-                         1 + loss_value]
-            feed_dict = {p: v for (p, v) in zip(placeholders, in_values)}
+            in_values = x_batch_chunks + y_batch_chunks + [loss_value]
+            if x_batch.shape[0] != dataset_params['batch_size']:
+                s_placeholders = (placeholders[0][:s_gpus['train']] +
+                                  placeholders[1][:s_gpus['train']] +
+                                  [placeholders[2]])
+                feed_dict = {p: v for(p, v) in zip(s_placeholders,
+                                                   in_values)}
+            else:
+                f_placeholders = (placeholders[0] + placeholders[1] +
+                                  [placeholders[2]])
+                feed_dict = {p: v for (p, v) in zip(f_placeholders, in_values)}
 
             # train_op does not return anything, but must be in the
             # outputs to update the gradient
             if cum_iter % cfg.train_summary_freq == 0:
+                if x_batch.shape[0] != dataset_params['batch_size']:
+                    t_outs = train_outs[1] + [train_summary_op[1]]
+                else:
+                    t_outs = train_outs[0] + [train_summary_op[0]]
                 loss_value, _, summary_str = cfg.sess.run(
-                    train_outs + [train_summary_op],
-                    feed_dict=feed_dict)
+                    t_outs, feed_dict=feed_dict)
+                sv.summary_computed(cfg.sess, summary_str)
+                # Write Mean_tower_loss summary
+                summary = tf.Summary.Value(tag='Mean_tower_loss_train',
+                                           simple_value=loss_value)
+                summary_str = tf.Summary(value=[summary])
```

Comment:
What is the reason why this has to be done here rather than in the `train_summary_op`?